.github/workflows/build_and_test_nix.yml

```diff
@@ -96,3 +96,32 @@ jobs:
       
     - name: Code Coverage
       run: bash <(curl -s https://codecov.io/bash) -X gcov -s ${{github.workspace}}/build_${{matrix.buildconfiguration}}
+
+  build_nix_deterministic_bundles:
+    name: Build and Test [${{matrix.os}},Configuration=0,Deterministic]
+    runs-on: ${{matrix.os}}
+    env:
+      BUILD_CONFIGURATION: 0
+      BUILD_WITH_DETERMINISTIC: 1
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-22.04, macOS-11, macos-12]
```

Comment:
are two macOS versions necessary? can we pick one?

Reply:
Sure. macos-12?

Reply:
the latest OS version sounds reasonable to me. Take a look at the github action documentation, there may be a moniker you can use for the latest macOS, like there is for Windows (windows-latest), so we don't have to keep changing the value in this file every time a new macOS version is supported by github actions.